zap/src/test/java/org/zaproxy/zap/model/TargetUnitTest.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2020 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.model;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+
+import org.junit.jupiter.api.Test;
+import org.parosproxy.paros.model.SiteNode;
+
+/** Unit test for {@link Target}. */
+class TargetUnitTest {
+
+    @Test
+    void shouldNotSetNullSiteNodeWhenConstructing() {
```

Comment:
These two test methods confuse me. They're named as if null shouldn't be accepted but then assert a null result.

Reply:
That's because they are expected to still be null, before the changes the (wrapped) site node would be added to the list of the start nodes which would make it non-null. The assert for the start node getter is just to exercising it, i.e. still works properly in this specific case.